tests/asserts/util.py

```diff
@@ -11,13 +11,24 @@ def get_publication_header(publication):
     return headers
 
 
-def run_action(publication, action):
+def run_action(publication, action, *, cache=None):
+    param_def = {
+        'headers': get_publication_header,
```

Comment:
Nice.

I am thinking about how we can cache more specific calls, e.g. get_publication_info for each key. We could use approach based on method signature (method + arguments and their values) as a key in dictionary you are calling `param_def` here. Then, we can pass this cache object to each assert method instead of specific named parameters. Not in this PR.